Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionGenericCEF.yaml

```diff
@@ -0,0 +1,302 @@
+        , EventStartTime= TimeGenerated // StartTime is empty on all our records
+    | lookup ProtocolLookup on $left.NumericProtocol==$right.Protocol
+    | extend
+        NetworkProtocol=coalesce(NetworkProtocol, toupper(Protocol))
+    // //
+        | extend hostelements=split(DeviceName,'.')
+        | extend DvcHostname=tostring(hostelements[0])
+               , DvcDomain=strcat_array( array_slice(hostelements,1,-1), '.')
+        | extend DvcFQDN = iff(DeviceName contains ".",DeviceName,"" )
+               , DvcDomainType=iff(DeviceName contains ".","FQDN","" )
+        | project-away hostelements
+    // //
+        | extend hostelements=split(SourceHostName,'.')
+        | extend SrcHostname=tostring(hostelements[0])
+               , SrcDomain=strcat_array( array_slice(hostelements,1,-1), '.')
+        | extend SrcFQDN = iff(SourceHostName contains ".",SourceHostName,"" )
+               , SrcDomainType=iff(SourceHostName contains ".","FQDN","" )
+        | project-away hostelements
+    // //
+        | extend hostelements=split(DestinationHostName,'.')
+        | extend DstHostname=tostring(hostelements[0])
+               , DstDomain=strcat_array( array_slice(hostelements,1,-1), '.')
+        | extend DstFQDN = iff(DestinationHostName contains ".",DestinationHostName,"" )
+               , DstDomainType=iff(DestinationHostName contains ".","FQDN","" )
+        | project-away hostelements
+    // //
+    | extend  DvcOriginalAction = DeviceAction, DeviceAction=tolower(DeviceAction)
+    | lookup  DeviceActions on $left.DeviceAction == $right.OriginalDeviceAction
+    |extend EventResult=iff(DvcAction =='Allow', 'Success', 'Failure' ) 
+          , NetworkDirection=case(CommunicationDirection=='0','Inbound',CommunicationDirection=='1','Outbound','')
+    | project-rename 
+         Dvc=DeviceName 
+      ,  DvcMacAddr=DeviceMacAddress
+      ,  DvcOutboundInterface=DeviceOutboundInterface
+      ,  DvcInboundInterface=DeviceInboundInterface
+      , DstBytes=ReceivedBytes
+      , DstMacAddr=DestinationMACAddress
+      , DstNatIpAddr=DestinationTranslatedAddress
+      , DstUserSid=DestinationUserID
+      , DstUsername=DestinationUserName
+      , DstIpAddr=DestinationIP
+      , DstPortNumber=DestinationPort
+      , DstNatPortNumber=DestinationTranslatedPort
+      ,  EventVendor=DeviceVendor
+      ,  EventProduct=DeviceProduct
+      ,  EventProductVersion=DeviceVersion 
+      ,  EventMessage=Message
+      ,  EventResourceId=_ResourceId
+      , NetworkApplicationProtocol=ApplicationProtocol
+      ,  SrcBytes=SentBytes
+      ,  SrcMacAddr=SourceMACAddress
+      ,  SrcNatIpAddr=SourceTranslatedAddress
+      ,  SrcUserSid=SourceUserID
+      ,  SrcIpAddr=SourceIP
+      ,  SrcPortNumber=SourcePort
+      ,  SrcNatPortNumber=SourceTranslatedPort
+      ,  SrcUsername=SourceUserName
+    | extend 
+        Dst=DstIpAddr
+        , EventEndTime=EventStartTime // Empty on all our records
```

Comment:
Actually TImeGenerated would be better, assuming start is not empty somewhere.

Reply:
Appologies, the comment is old. EventStartTime has since been changed to TimeGenerated